### ISSUE TITLE
Ray fails to serialize Torch tensor.

### ISSUE THREAD

**Issue by robertnishihara** [ray-project/ray#955]
I'm running into problems serializing pytorch tensors.

```python
import ray
import torch

ray.init()

x = torch.Tensor(10)

x_id = ray.put(x)

result = ray.get(x_id)
```

Inspecting the values shows that `x` and `result` are different.

```python
>>> x
 0.0000e+00
-8.5899e+09
 0.0000e+00
-8.5899e+09
 5.6052e-45
 1.4714e-43
 1.5975e-43
 1.4153e-43
 1.3873e-43
 1.6255e-43
[torch.FloatTensor of size 10]

>>> result
[torch.FloatTensor with no dimension]

>>> result[0]
RuntimeError: dimension 1 out of range of 0D tensor at /Users/soumith/code/builder/wheel/pytorch-src/torch/lib/TH/generic/THTensor.c:24
```

cc @vitchyr

**Comment by robertnishihara** [comment]
The fix for this is waiting on #1138.

**Comment by robertnishihara** [comment]
No longer waiting on #1138 because it is fixed in pyarrow, so #1152 fixes it.

**Comment by robertnishihara** [comment]
Does it work with earlier versions of pytorch? Could it be the version update (I assume this is pytorch 0.4)?

It's possible that simply switching to the latest Arrow will fix the issue because of https://github.com/apache/arrow/commit/df9bdc10a9481ee328e22cf0dc20e17d4ca4f8d0.

**Comment by richardliaw** [comment]
sure; how do I try that out? 

cc @alok

**Comment by alok** [comment]
@richardliaw If you use the code snippet I put in #1856 and the version of arrow Robert mentioned (or manually change your local copy of the arrow code since it's a one line fix), it should work. I just tested your code snippet in this issue with the custom serializer and it works fine.

**Comment by richardliaw** [comment]
Is it possible to get away without custom serializers?

**Comment by alok** [comment]
Only if you use that arrow PR. I use the custom serializer for local testing and the arrow PR just makes it the official solution.

**Comment by robertnishihara** [comment]
@richardliaw the line to change is

https://github.com/ray-project/ray/blob/78bb887dd4c375a3e8141cdad3cefe23b68244a9/thirdparty/scripts/build_arrow.sh#L50

**Comment by alok** [comment]
This can be closed since #2100 was merged.
